sphinxcontrib/doxylink/doxylink.py

```diff
@@ -354,6 +354,25 @@ def find_doxygen_link(name, rawtext, text, lineno, inliner, options={}, content=
     return find_doxygen_link
 
 
+def download_file(app, url, output_location):
+    if not os.path.isabs(output_location):
+        output_location = os.path.join(app.outdir, output_location)
+    if os.path.exists(output_location):
+        return
+    response = requests.get(url, allow_redirects=True)
+    if response.status_code != 200:
+        report_warning(app.env,
+                       standout("Could not find file %s. Make sure your `doxylink_remote_pdf_files` config variable is "
+                                "set correctly." % url))
+        return
+    os.makedirs(os.path.dirname(output_location), exist_ok=True)
+    with open(output_location, 'wb') as file:
+        file.write(response.content)
+
+
 def setup_doxylink_roles(app):
     for name, (tag_filename, rootdir) in app.config.doxylink.items():
         app.add_role(name, create_role(app, tag_filename, rootdir, name))
+        if rootdir in app.config.doxylink_remote_pdf_files:
```

Comment:
check that url (rootdir) truly is url and not something strange? Well in practice it could be local path as well but you should verify that it is not blank at least

Reply:
Good call. This will be more clear than the error message from the requests module: 01a01ae30febac5dc876f6937be582abe6b2bef2

Reply:
I think that `rootdir` is allowed to be an empty string when Sphinx' output directory is the same as the root of Doxygen's html documentation.